### ISSUE TITLE
Making CRLs required and OCSP Optional

### ISSUE THREAD

**Issue by dougbeattie** [cabforum/servercert#505]
This Ballot:

- Ballot SC-063 V4: Make OCSP Optional, Require CRLs, and Incentivize Automation

didn't quite specify CRLs as being required, if that was the full intent of the ballot.

If we want to mandate CRLs, then don't we need to mandate CDP in subscriber certificates (except for shore lived certificates)?

For example, https://github.com/cabforum/servercert/blob/a0efd83d3818fe5c3df23bf4b32483cc4e6f133c/docs/BR.md#712112-crl-distribution-points  says:

The CRL Distribution Points extension MUST be present in:
- Subordinate CA Certificates; and
- Subscriber Certificates that 1) do not qualify as "Short-lived Subscriber Certificates" and 2) do not include an Authority Information Access extension with an id-ad-ocsp accessMethod.

Which means you can omit CDP if you have id-ad-ocsp accessMethod.  Is that the intent of the ballot making CRLs required?

Also you could interpret these sections as the CDP can optional.  
  https://github.com/cabforum/servercert/blob/a0efd83d3818fe5c3df23bf4b32483cc4e6f133c/docs/BR.md#712112-crl-distribution-points

https://github.com/cabforum/servercert/blob/a0efd83d3818fe5c3df23bf4b32483cc4e6f133c/docs/BR.md#71276-subscriber-certificate-extensions

On the other hand, if the intent of the ballot is to ensure that every TLS CA has a CRL, but CDP is not required, then we should make that clear.

**Comment by clintwilson** [comment]
I think your reading of the requirements is correct, excepting that we _don't_ need to mandate CDPs in order to mandate CRLs.
There are 2 separate requirements interacting here, aiui.

1. CAs must issue CRLs which collectively represent all revoked certificates under that CA. This has been a Root Program requirement already, now codified in the TBRs. This requirement doesn't impact certificate profiles or content as CRLs can be issued regardless of whether there is a CDP in the certificates for which the CRLs are authoritative and still function perfectly well as a CRL.
2. CAs must include a pointer to at least one certificate status information source. This can be a CRL or OCSP.

That is, from my reading it already is clear that the ballot 1) requires CRLs by ensuring every TLS CA issues a CRL, and does not impose a requirement that TLS CAs include a CDP, while introducing the allowance for the inclusion of a CDP to obviate the need to include an OCSP pointer, thus 2) making OCSP optional.

**Comment by XolphinMartijn** [comment]
@clintwilson I agree with your explanation here. However, having another read of the section made me realize there's a potential timing issue.

Taking the "CAs must issue CRLs which collectively represent all revoked certificates under that CA. " statement, which indeed the BRs state, what would be the requirement is the CA has not yet issued any certificate? I would assume the best reading here would mean the scope is 0 certificates, but a CRL must still be issued (be it with no revokedCertificates extension).

That however, poses an issue when a CA wants to issue a new SubCA. One could argue that the moment a new SubCA is issued, that CRL must be issued at the same time. If not, the SubCA would be out of compliance. 

Yet, issuing these at the same point is, I think fair to say, not feasible, nor desired. CA's may want to make sure the SubCA is properly setup after being issued, before enabling any signing operations. 

As such, I wonder if we should improve this section by taking the approach that the Mozilla Root Store Policy has taken, by rephrasing the 5th paragraph in section 7.2 to something akin to (emphasis mine):

_Minimally, CAs MUST issue either a "full and complete" CRL or a set of "partitioned" CRLs which cover the complete set of Certificates issued by the CA **within 7 days of such CA issuing its first certificate**. In other words, if issuing only partitioned CRLs, the combined scope of those CRLs must be equivalent to that of a full and complete CRL.;_

(Or a shorter timeframe if so desired)

**Comment by dzacharo** [comment]
Notes from the 2024-12-05 SCWG Teleconference:
- Issuance of CRLs are required by the BRs but CRLDP is not required to be included in end-entity certificates
- This should be clarified along with proper timing after creating a CA Certificate
- Doesn't look like a cleanup item but could be considered one because it loosens an instantaneous requirement. Clint and Martijn agree on this. Trev wants to align with policies to require this information in CCADB.

**Comment by clintwilson** [comment]
I'll note that the Apple and Mozilla policies require disclosure of CRLs to the CCADB within 7 days after a CAs first issuance; if we align the BRs exactly with that, it may introduce a similar timing issue if CAs generate a CRL at the end of that 7 day period. Perhaps it would be better to instead require generation of the CRL (in the BRs) within 5-6 days, to allow some gap between generation and disclosure.
Or we can do 7 days and just expect CAs to manage both tasks within that timeframe 🤷‍♀️

**Comment by XolphinMartijn** [comment]
@clintwilson I've taken the initial proposed text with 7 days. Yes, I see your concern, but I'd argue that it is easily solvable for a CA by "just disclosing the CRL in time". A CCADB disclosure doesn't take more than a few minutes.
